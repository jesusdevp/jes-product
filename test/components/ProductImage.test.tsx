import React from 'react';
import renderer from 'react-test-renderer'
import { ProductImage } from '../../src/components/ProductImage';
import { ProductCard } from '../../src/components/ProductCard';
import { product2 } from '../data/products';

describe('ProductImage', () => { 

    test('Debe mostrar con la url por props', () => { 
        const wrapper = renderer.create(
            <ProductImage img='https://' />
        )

        expect( wrapper.toJSON() ).toMatchSnapshot()

     })

     test('Debe mostrar la imagen del producto', () => {

        const wrapper = renderer.create(
            <ProductCard product={ product2 } >
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot()

     })

  })