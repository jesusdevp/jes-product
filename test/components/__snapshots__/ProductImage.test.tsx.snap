// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar con la url por props 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="https://"
/>
`;

exports[`ProductImage Debe mostrar la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
